homeassistant/components/fastdotcom/sensor.py

```diff
@@ -1,55 +1,49 @@
 """Support for Fast.com internet speed testing sensor."""
+from __future__ import annotations
+
+from typing import Any
+
 from homeassistant.components.sensor import SensorEntity
 from homeassistant.const import DATA_RATE_MEGABITS_PER_SECOND
-from homeassistant.core import callback
+from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
 from . import DATA_UPDATED, DOMAIN as FASTDOTCOM_DOMAIN
 
 ICON = "mdi:speedometer"
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
     """Set up the Fast.com sensor."""
     async_add_entities([SpeedtestSensor(hass.data[FASTDOTCOM_DOMAIN])])
 
 
 class SpeedtestSensor(RestoreEntity, SensorEntity):
     """Implementation of a FAst.com sensor."""
 
-    def __init__(self, speedtest_data):
+    def __init__(self, speedtest_data: dict[str, Any]) -> None:
         """Initialize the sensor."""
-        self._name = "Fast.com Download"
-        self.speedtest_client = speedtest_data
-        self._state = None
+        self._attr_name: str = "Fast.com Download"
+        self._attr_unit_of_measurement: str = DATA_RATE_MEGABITS_PER_SECOND
+        self._attr_icon: str = ICON
+        self._attr_should_poll: bool = False
```

Comment:
These are typed on the base class, it is better not to add typing for them, also since all are constants you can put them class variables, like this above `__init__`
```suggestion
        _attr_name = "Fast.com Download"
        _attr_unit_of_measurement = DATA_RATE_MEGABITS_PER_SECOND
        _attr_icon = ICON
        _attr_should_poll = False
```